src/node/internal/internal_timers_promises.ts

```diff
@@ -62,16 +64,23 @@ export async function setTimeout<T = void>(
   }
 
   const { promise, resolve, reject } = Promise.withResolvers<T>();
+  let onCancel: OnCancelCallback;
 
   const timer = timers.setTimeout(() => {
     resolve(value as T);
-  }, delay ?? 0);
+    if (onCancel) {
+      signal?.removeEventListener('abort', onCancel);
+    }
```

Comment:
Another change that might be worthwhile... on the `addEventListener('abort', ...)` call, it would probably be a good idea to add the `once: true` option..

Reply:
Yeah agreed. Applying it now.